### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by amartya-rs**
feat: home page of ecom site
### Files to check

- [src/assets/icons.js ] 
- [src/components/footer/Footer.jsx ] 
- [ src/components/homepage-card/HomePageCard.jsx] 
- [ src/context/cart-context.js] 
- [ src/pages/home-page/HomePage.jsx]

### CODE REVIEW

File: src/context/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { useEffect, createContext, useContext, useReducer } from "react";
+import axios from "axios";
+
+const CartContext = createContext();
```
Comment (archit7448) [inline src/context/cart-context.js:4]: in this you are using for taking product data you can give product provider , it creates confusion I misunderstood , it is  related for Cart data

**Review by archit7448 — APPROVED** [review]
File structure proper code is cleaner Looks good to merge

File: src/context/cart-context.js (hunk 1 above)
Comment (amartya-rs) [reply src/context/cart-context.js:4]: Actually I am using only 1 provider; product data, cart data etc all will be in this same provider.

**Review by srishtigautam21 — APPROVED** [review]
The code is clean and readable looks good to merge

File: src/context/cart-context.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   useEffect(() => {
+      (async () => {
+         try {
+            const response = await axios.get("/api/products");
+            dispatch({
+               type: "UPDATE_DATA",
+               payload: response.data.products,
+            });
+         } catch (error) {
+            console.log(error);
```
Comment (purvasharmadev) [inline src/context/cart-context.js:17]: fix: remove console.log

**Review by purvasharmadev — APPROVED** [review]
The code is looking clean and understandable, Folder structure is good too.
Good to go.

Suggestion: add  preview links for PRs

**Comment by santhoshpemmaka** [comment]
The code looks good and clean 💯 .  Good to merge 👍

**Review by santhoshpemmaka — APPROVED** [review]

File: src/context/cart-context.js (hunk 2 above)
Comment (amartya-rs) [reply src/context/cart-context.js:17]: Will remove it, thanks for pointing it out

**Pull request merged**